activesupport/CHANGELOG.md

```diff
@@ -1,3 +1,26 @@
+*   Allow entirely opting out of deprecation warnings
+
+    Previously if you did `app.config.active_support.deprecation = :silence`, some work would
+    still be done on each call to `ActiveSupport::Deprecation.warn`. In very hot paths, this could
+    cause performance issues.
+
+    Now, you can make `ActiveSupport::Deprecation.warn` a no-op:
+
+    ```ruby
+    config.active_support.report_deprecations = false
```

Comment:
Can you please document in the configuration guide. Also, I think this should be default in production for new apps.

Reply:
Updated both. Note that the production defaults used to be `log` (not `silence`) so this would be a functional change for new apps (not just a performance change).